### ISSUE TITLE
Error "file name too long" if DOCKER_CONFIG defined and path more 105 characters. Podman 3.0.1

### ISSUE THREAD

**Issue by sappersd** [containers/podman#9634]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**
If I use VOLUME instruction into Dockerfile and set env DOCKER_CONFIG with path length more 105 characters - I get error 'file name too long'.
It relate with authfile. But if I additionaly set REGISTRY_AUTH_FILE more 105 characters - it works without errors.

Also there are log spam like:
`time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"`
But it looks like there was fixed:
Setting DOCKER_CONFIG leads to 20 line authfile log spam on INFO log level #9473

**Output of `podman version`:**
```
bash-4.2# podman version
Version:      3.0.1
API Version:  3.0.0
Go Version:   go1.15.5
Built:        Fri Feb 19 20:35:26 2021
OS/Arch:      linux/amd64
bash-4.2# uname -a
Linux 04ee2482f4ee 5.9.3-1.el7.elrepo.x86_64 #1 SMP Sat Oct 31 09:41:18 EDT 2020 x86_64 x86_64 x86_64 GNU/Linux
```
**testcase:**
1. create Dockerfile
```
bash-4.2# echo '
> FROM centos:7.8.2003 as base
> RUN mkdir /myvol
> RUN echo "hello world" > /myvol/greeting
> VOLUME /myvol
> ' >  Dockerfile
bash-4.2# unset REGISTRY_AUTH_FILE
bash-4.2# unset DOCKER_CONFIG
```
2. run build with DOCKER_CONFIG  with path length 105 chars.
build OK
``` 
bash-4.2# export DOCKER_CONFIG=/tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
bash-4.2# podman build -t test1 .
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/1
STEP 1: FROM centos:7.8.2003 AS base
STEP 2: RUN mkdir /myvol
--> Using cache 6e4745bd5141b656b764adfea9f8051fa4ba5d662f8772d5bf561a243b36673e
--> 6e4745bd514
STEP 3: RUN echo "hello world" > /myvol/greeting
--> Using cache ff7fd8d030f79c4240e44514bf48179db8907798d4f8ce59ba6edbe35d72e3cf
--> ff7fd8d030f
STEP 4: VOLUME /myvol
--> Using cache 2fc5fc8e06e4c109d5bfef5bbb65272b11267c439e77a66d1ef222901aa657c7
STEP 5: COMMIT test1
--> 2fc5fc8e06e
2fc5fc8e06e4c109d5bfef5bbb65272b11267c439e77a66d1ef222901aa657c7
```

3. try 106 chars into path
get error
```
bash-4.2# export DOCKER_CONFIG=/tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
bash-4.2# podman build -t test1 .
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
INFO[0000] Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
STEP 1: FROM centos:7.8.2003 AS base
STEP 2: RUN mkdir /myvol
--> Using cache 6e4745bd5141b656b764adfea9f8051fa4ba5d662f8772d5bf561a243b36673e
--> 6e4745bd514
STEP 3: RUN echo "hello world" > /myvol/greeting
--> Using cache ff7fd8d030f79c4240e44514bf48179db8907798d4f8ce59ba6edbe35d72e3cf
--> ff7fd8d030f
STEP 4: VOLUME /myvol
Error: error building at STEP "VOLUME /myvol": stat /var/lib/containers/storage/overlay/e88ec1fb57d899524b7dc7a63a049a02ab3037a4ec8f6a91e048c565b3f83203/merged/time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
time="2021-03-05T14:56:50Z" level=info msg="Using DOCKER_CONFIG environment variable for authfile path /tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12"
/myvol: file name too long
bash-4.2#
```

4. set REGISTRY_AUTH_FILE with 106 path length
build OK
and there are no "info" spam
```
bash-4.2# export REGISTRY_AUTH_FILE=/tmp/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/123456789/12
bash-4.2# podman build -t test1 .
STEP 1: FROM centos:7.8.2003 AS base
STEP 2: RUN mkdir /myvol
--> Using cache 6e4745bd5141b656b764adfea9f8051fa4ba5d662f8772d5bf561a243b36673e
--> 6e4745bd514
STEP 3: RUN echo "hello world" > /myvol/greeting
--> Using cache ff7fd8d030f79c4240e44514bf48179db8907798d4f8ce59ba6edbe35d72e3cf
--> ff7fd8d030f
STEP 4: VOLUME /myvol
--> Using cache 2fc5fc8e06e4c109d5bfef5bbb65272b11267c439e77a66d1ef222901aa657c7
STEP 5: COMMIT test1
--> 2fc5fc8e06e
2fc5fc8e06e4c109d5bfef5bbb65272b11267c439e77a66d1ef222901aa657c7
bash-4.2#
```

**Output of `podman info --debug`:**

```
bash-4.2# podman info --debug
host:
  arch: amd64
  buildahVersion: 1.19.4
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.26-2.el7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.26, commit: 7077467ffea93b7fe6e5ebeece408c8c10aaad92'
  cpus: 6
  distribution:
    distribution: '"centos"'
    version: "7"
  eventLogger: file
  hostname: 04ee2482f4ee
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.9.3-1.el7.elrepo.x86_64
  linkmode: dynamic
  memFree: 16978096128
  memTotal: 73793585152
  ociRuntime:
    name: runc
    package: containerd.io-1.4.3-3.1.el7.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc92
      commit: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
      spec: 1.0.2-dev
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    selinuxEnabled: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 0
  swapTotal: 0
  uptime: 2956h 31m 57.68s (Approximately 123.17 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.imagestore: /var/lib/shared-cache/storage
    overlay.mountopt: nodev
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 111
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.0.0
  Built: 1613766926
  BuiltTime: Fri Feb 19 20:35:26 2021
  GitCommit: ""
  GoVersion: go1.15.5
  OsArch: linux/amd64
  Version: 3.0.1
```

**Comment by rhatdan** [comment]
Yes once we have this vendored in as a fix it will be fixed in podman 3.1

**Comment by mheon** [comment]
@rhatdan Can I close, given this is fixed upstream?
